config/v1/types_operatorhub.go

```diff
@@ -0,0 +1,84 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// OperatorHubKind is the string value of OperatorHub kind
+var OperatorHubKind = "OperatorHub"
+
+// OperatorHubSpec defines the desired state of OperatorHub
+// +k8s:openapi-gen=true
+type OperatorHubSpec struct {
+	// HubSources is the list of default OperatorSources and their configuration
+	HubSources []HubSource `json:"hubSources,omitempty"`
+}
+
+// OperatorHubStatus defines the observed state of OperatorHub
+// +k8s:openapi-gen=true
+type OperatorHubStatus struct {
+	// HubSourcesStatus encapsulates the result applying the configuration
+	HubSourcesStatus []HubSourceStatus `json:"hubSourcesStatus,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OperatorHub is the Schema for the operatorhubs API
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +genclient:nonNamespaced
+type OperatorHub struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OperatorHubSpec   `json:"spec,omitempty"`
+	Status OperatorHubStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OperatorHubList contains a list of OperatorHub
+type OperatorHubList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []OperatorHub `json:"items"`
+}
+
+// HubSource is used to specify the OperatorSource and its configuration
+type HubSource struct {
+	// Name is the name of one of the default OperatorSources
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+	// Disable is used to disable a default OperatorSource on cluster
+	Disable bool `json:"disable"`
+}
+
+// HubSourceStatus is used to reflect the current state of applying the
+// configuration to a default source
+type HubSourceStatus struct {
+	// Name is the name of one of the default OperatorSources
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name,omitempty"`
+	// Configuration is the state of the default OperatorSources configuration
+	Configuration map[string]string `json:"configuration,omitempty"`
+	// Condition indicates success or failure in applying the configuration
+	Condition string `json:"condition,omitempty"`
+	// Message provides more information regarding failures
+	Message string `json:"message,omitempty"`
+}
+
+// EnsureGVK sets the group, version, and kind strings from the internal
+// reference that we defined in the v1 package. The object the sdk client
+// returns does not set these so we must find the correct values and set them
+// manually.
+func (o *OperatorHub) EnsureGVK() {
```

Comment:
I don't think that this helper function should be here, I might be wrong but I have not seen another type with this type of function.

Reply:
I am fine removing it if that is convention. 

My reason for keeping it here is that it allows us to do `operatorhub.EnsureGVK()` for an `OperatorHub` object which seems a natural way of doing things. This can't be done by as a helper function elsewhere.

Reply:
+1

Doesn't belong here. Also operator sdk with controller-runtime client does not need seeting TypeMeta anymore.